MapCore/src/main/java/org/wwarn/mapcore/client/components/customwidgets/facet/FacetSearchableCheckBoxWidget.java

```diff
@@ -284,7 +280,56 @@ public void onSuggestionsReady(SuggestOracle.Request request, SuggestOracle.Resp
             updatePanelList(suggestedList);
         }
     };
+    @UiField
+    HTMLPanel heading;
+
+    @UiField
+    SpanElement headingValue;
+
+    @UiField
+    Anchor toggleFilter;
+
+    @UiHandler("clearSelectionControl")
+    public void clearSelectionHandleClick(ClickEvent event) {
+        event.preventDefault();
+        unSelectAndReset();
+        //TODO: resolve this hack
+        // fire value change event inform client that reset was completed
+        ValueChangeEvent.fire(checkBoxList.iterator().next(), true);
+    }
 
+    @UiField
+    TextBox searchBox;
+    @UiField
+    ScrollPanel scrollpanel;
+    @UiField
+    FlowPanel filterMainBody;
+    @UiField
+    SpanElement toggleFilterField;
+
+    @UiHandler("toggleFilter")
+    public void handleClick(ClickEvent event) {
+        event.preventDefault();
+        filterMainBody.setVisible(!filterMainBody.isVisible());
+        updateToggleFilterView();
+    }
+
+    @NotNull
+    private String getToggleFilterValue() {
+        return (filterMainBody.isVisible())?"Hide":"Show";
+    }
+    private void updateToggleFilterView() {
+//        toggleFilter.setHTML(getToggleFilterValue());
+        toggleFilter.setTitle(getToggleFilterValue());
+        toggleFilterField.setTitle(getToggleFilterValue());
+        if(filterMainBody.isVisible()) {
+            toggleFilterField.removeClassName("glyphicon-plus");
+            toggleFilterField.addClassName("glyphicon-minus");
```

Comment:
Just to let you know, If you use the glyphicon class, that will force the Surveyor framework to use the bootstrap library. I will be more than happy to make this assumption, because I was looking for use some icons in the surveyor framework.